packages/lab/src/date-input/DateInputSingle.tsx

```diff
@@ -224,27 +200,21 @@ export const DateInputSingle = forwardRef<HTMLDivElement, DateInputSingleProps>(
       name: "DateInputSingle",
       state: "dateValue",
     });
-    const preservedTime = useRef<TimeFields | undefined>(
-      extractTimeFieldsFromDate(date || null),
-    );
-
-    const format = useCallback(
-      (date: DateValue | null) => {
-        return formatProp
-          ? formatProp(date)
-          : defaultFormatDate(date, locale, { timeZone });
-      },
-      [formatProp, locale, timeZone],
-    );
+    const lastAppliedValue = useRef<string>(dateValue);
+    const preservedTime = useRef<TimeFields | null>(null);
+    preservedTime.current = date ? dateAdapter.getTime(date) : null;
 
     // Update date string value when selected date changes
     useEffect(() => {
-      const formattedDate = format(date || null);
-      if (formattedDate) {
-        setDateValue(formattedDate);
-        onDateValueChange?.(formattedDate, true);
+      if (date && dateAdapter.isValid(date)) {
+        const formattedValue = dateAdapter.format(date, format, locale);
+        const hasValueChanged = formattedValue !== dateValue;
+        if (hasValueChanged) {
```

Comment:
This prevents controlling `selectedDate` to `null` to reset input to empty. As Kris suggested, maybe consider using "invalid" date in `onSelectionChange`'s `date`, so there will be 4 distinct choices
- `undefined` (controlled vs uncontrolled)
- `null`, controlled with nothing
- invalid date, controlled with date but invalid
- valid date, controlled with date and also valid

Reply:
done